### PR TITLE
SP_for_API1600

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Added extended support for Server profile resource to OV API 1600. 

### Issues Resolved
N/A

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/shared_samples/server_profile.rb [hunk 1]
```diff
@@ -71,6 +71,7 @@
 item3.retrieve!
 puts "\nServer Profile updated successfully! Name: #{item3['name']}"
 
+# This method #get_availble_server is removed for  Oneview REST API Version 1600.
```
Comment (soodpr) [inline examples/shared_samples/server_profile.rb:74]: better to say "This method supports till OneView REST API Version 1200" and add begin and rescue block for this endpoint

File: examples/shared_samples/server_profile.rb [hunk 2]
```diff
@@ -109,6 +110,7 @@
 item2.update_from_template
 puts "\nServer Profile updated successfully!"
 
+# This method #get_profile_template is removed for  Oneview REST API Version 1600.
```
Comment (soodpr) [inline examples/shared_samples/server_profile.rb:113]: same as above

File: lib/oneview-sdk/resource/api1600/c7000/server_profile.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::Client] client The client object for the OneView appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        def initialize(client, params = {}, api_ver = nil)
+          @data ||= {}
+          # Default values
+          @data['type'] ||= 'ServerProfileV12'
+          super
+        end
```
Comment (soodpr) [inline lib/oneview-sdk/resource/api1600/c7000/server_profile.rb:28]: add unavailable_method for the removed endpoints

File: examples/shared_samples/server_profile.rb (hunk 1 above)
Comment (AsisBagga) [reply examples/shared_samples/server_profile.rb]: @soodpr Updated.

File: examples/shared_samples/server_profile.rb (hunk 2 above)
Comment (AsisBagga) [reply examples/shared_samples/server_profile.rb]: @soodpr Updated.

File: lib/oneview-sdk/resource/api1600/c7000/server_profile.rb (hunk 3 above)
Comment (AsisBagga) [reply lib/oneview-sdk/resource/api1600/c7000/server_profile.rb:28]: @soodpr Updated.

File: examples/shared_samples/server_profile.rb [hunk 4]
```diff
@@ -71,6 +71,7 @@
 item3.retrieve!
 puts "\nServer Profile updated successfully! Name: #{item3['name']}"
 
+# This method supports till OneView REST API Version 1200
```
Comment (soodpr) [inline examples/shared_samples/server_profile.rb:74]: still begin and rescue block is missing here.

Comment (AsisBagga) [reply examples/shared_samples/server_profile.rb:74]: @soodpr its there. Unavailable method for these methods are added to library class for SP for API1600 and begin rescue is added in example file.

Comment (soodpr) [reply examples/shared_samples/server_profile.rb:74]: its not present at line# 75. Please add it

Comment (AsisBagga) [reply examples/shared_samples/server_profile.rb:74]: @soodpr Its added now.

**Pull request merged**